src/geo_key_directory.rs

```diff
@@ -0,0 +1,647 @@
+use num_enum::{IntoPrimitive, TryFromPrimitive};
+use tiff::tags::Tag;
+use tiff::{TiffError, TiffFormatError, TiffResult};
+
+#[derive(Debug, PartialEq)]
+pub struct GeoKeyDirectory {
+    pub key_directory_version: u16,
+    pub key_revision: u16,
+    pub minor_revision: u16,
+    pub model_type: Option<u16>,
+    pub raster_type: Option<u16>,
+    pub citation: Option<String>,
+    pub geographic_type: Option<u16>,
+    pub geog_citation: Option<String>,
+    pub geog_geodetic_datum: Option<u16>,
+    pub geog_prime_meridian: Option<u16>,
+    pub geog_linear_units: Option<u16>,
+    pub geog_linear_unit_size: Option<f64>,
+    pub geog_angular_units: Option<u16>,
+    pub geog_angular_unit_size: Option<f64>,
+    pub geog_ellipsoid: Option<u16>,
+    pub geog_semi_major_axis: Option<f64>,
+    pub geog_semi_minor_axis: Option<f64>,
+    pub geog_inv_flattening: Option<f64>,
+    pub geog_azimuth_units: Option<u16>,
+    pub geog_prime_meridian_long: Option<f64>,
+    pub projected_type: Option<u16>,
+    pub proj_citation: Option<String>,
+    pub projection: Option<u16>,
+    pub proj_coord_trans: Option<u16>,
+    pub proj_linear_units: Option<u16>,
+    pub proj_linear_unit_size: Option<f64>,
+    pub proj_std_parallel1: Option<f64>,
+    pub proj_std_parallel2: Option<f64>,
+    pub proj_nat_origin_long: Option<f64>,
+    pub proj_nat_origin_lat: Option<f64>,
+    pub proj_false_easting: Option<f64>,
+    pub proj_false_northing: Option<f64>,
+    pub proj_false_origin_long: Option<f64>,
+    pub proj_false_origin_lat: Option<f64>,
+    pub proj_false_origin_easting: Option<f64>,
+    pub proj_false_origin_northing: Option<f64>,
+    pub proj_center_long: Option<f64>,
+    pub proj_center_lat: Option<f64>,
+    pub proj_center_easting: Option<f64>,
+    pub proj_center_northing: Option<f64>,
+    pub proj_scale_at_nat_origin: Option<f64>,
+    pub proj_scale_at_center: Option<f64>,
+    pub proj_azimuth_angle: Option<f64>,
+    pub proj_straight_vert_pole_long: Option<f64>,
+    pub vertical: Option<u16>,
+    pub vertical_citation: Option<String>,
+    pub vertical_datum: Option<u16>,
+    pub vertical_units: Option<u16>,
+}
+
+impl Default for GeoKeyDirectory {
+    fn default() -> Self {
+        Self {
+            key_directory_version: 1,
+            key_revision: 1,
+            minor_revision: 1,
```

Comment:
Assumes GeoTIFF version 1.1 as default, should be ok.

See requirements 2.9 under https://docs.ogc.org/is/19-008r4/19-008r4.html#_requirements_class_geokeydirectorytag:

> 
> http://www.opengis.net/spec/GeoTIFF/1.1/req/GeoKeyDirectoryTag.minorRevisionValue
> The MinorRevision for this standard SHALL be 0 or 1.
> 
> 0 = GeoTIFF 1.0 version. Accepted for existing files (GeoTIFF 1.0) that do not use values / codes that are suppressed in this version
> 
> 1 = this version 1.1. Recommended for production / writing a GeoTIFF file according to this standard (and developing software). Shall be used for files keys and codes not covered by GeoTIFF 1.0 (e.g. VerticalGeoKey).

Reply:
I will put that in the doc-Comment of `GeoKeyDirectory`.